daemon/import.go

```diff
@@ -60,15 +62,24 @@ func (daemon *Daemon) ImportImage(src string, repository, tag string, msg string
 		rc = inConfig
 	} else {
 		inConfig.Close()
+		if len(strings.Split(src, "://")) == 1 {
+			src = "http://" + src
+		}
 		u, err := url.Parse(src)
 		if err != nil {
 			return err
 		}
-		if u.Scheme == "" {
-			u.Scheme = "http"
-			u.Host = src
-			u.Path = ""
+
+		// Checking the file referenced by the URL exists to avoid sending
+		// status code 200 to the client although the download process will fail
+		resp, err = http.Head(u.String())
+		if err != nil {
+			return err
+		}
+		if resp.StatusCode >= http.StatusBadRequest {
```

Comment:
OK I have tested it. It seems like the `http.Head` automatically check the redirected resource.
For example, I set a HTTP server that returns the following header:
```
HTTP/1.1 301 Moved Permanently
Location: /my_file/file.tar
Content-Length: 0
Content-Type: text/plain; charset=utf-8
```
Which was successfully downloaded.
However, when I removed the `Location` header, I got 500 in the response and the following error message in the body:
```
{"message":"Head http://172.17.0.1:9871/my_image: 301 response missing Location header"}
```

So it seems like 3xx is "covered".